bakta_queries.py

```diff
@@ -35,61 +35,58 @@ def get_gene_info_by_name(cursor, gene_name, column):
     cursor.execute(query, (gene_name,))
     return cursor.fetchall()
 
-def extract_gene_sequence_from_fasta(fasta_file_path, contig_id, start, end, strand):
+def extract_gene_sequence_from_fasta(fasta_file_path, contig_id, start, stop, strand):
     """
     Extract the gene sequence from a FASTA file based on coordinates and strand.
     """
-    gene_sequence = ""
-    for record in SeqIO.parse(fasta_file_path, "fasta"):
-        if record.id == contig_id:
+    gene_sequence = None
+
+    for record in read(fasta_file_path, format='fasta'):
+        if record.metadata['id'] == contig_id:
             if strand == "1":
-                gene_sequence = record.seq[start:end]  # Adjust for 0-based indexing
+                gene_sequence = str(record[start:stop])
             elif strand == "-1":
-                gene_sequence = record.seq[start:end].reverse_complement()
+                gene_sequence = str(DNA(record[start:stop]).reverse_complement())
             break
-    return gene_sequence
+    if gene_sequence is not None:
+        return gene_sequence
+    else:
+        raise KeyError(f"{contig_id} not found in {fasta_file_path}")
 
-def write_gene_sequences_to_fasta(output_file_path, gene_sequences):
+def write_gene_sequences_to_fasta(output_file_path, genes):
     """
-    Write gene sequences to an output FASTA file.
+    Write gene sequence to FASTA file.
     """
-    with open(output_file_path, "w") as output_file:
-        output_file.write(gene_sequences)
+    with open(output_file_path, 'w') as output_file:
+        for gene_sequence in genes:
+            # Assuming gene_sequence is a string containing the sequence
+            output_file.write(gene_sequence)
+            output_file.write('\n')  # Add a newline to separate sequences
```

Comment:
if `gene_sequence` is a `str` then this is not FASTA format?

Reply:
Can we use the `skbio` writer methods?

Reply:
Yeah, I tried to do this originally, but I can't figure out how to modify the name of each sequence where the header is ">{gene_id}"

Reply:
I recommend `write_gene_sequences_to_fasta` take an iterable like a list of `skbio.DNA` objects. The `.write` method of each object can be used with a provided file. The example below uses a file-like object, but the semantics are consistent with a file

```python
>>> import skbio
>>> example = skbio.DNA('AATTGG', metadata={'id': 'foo'})
>>> import io
>>> x = io.StringIO()
>>> example.write(x)
<_io.StringIO object at 0x7fe58807b4c0>
>>> x.seek(0)
0
>>> x.read()
'>foo\nAATTGG\n'
```